lib/devise-security/patches/secure_validatable_information.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module DeviseSecurity::Patches
+  module SecureValidatableInformation
+    extend ActiveSupport::Concern
+
+    included do
+      alias_method(
+        :original_set_minimum_password_length, :set_minimum_password_length
+      )
+
+      define_method :set_minimum_password_length do
+        if devise_mapping.secure_validatable?
+          @minimum_password_length = resource_class.password_length.min
+          @minimum_password_complexity = resource_class.password_complexity
```

Comment:
I'm not sure why we would need to expose this?  The `@minimum_password_length` is used in some of the default views, so I can understand that, but the password complexity is not.

Isn't it just easier to directly access the configurations for `password_length` and `password_complexity` in custom views?

Reply:
The idea from https://github.com/devise-security/devise-security/issues/290 was for the same reason that password length is exposed for views, if password complexity was exposed then people could default use them in their views in the same fashion. 

Do you think it would be better to just support `@minimum_password_length` for devise purposes and then note if you want to access password complexity in your views that you should use the config directly instead?

Reply:
Let me just add that while i do agree that it is easier to access `password_length` and `password_complexity` in the views directly. However in this scenario the views also need to check if `secure_validatable` is used.

This is why I did prefer to use the same approach as devise does. Just to keep the inner workings out of the views.

Reply:
@olbrich bump on your opinions for this?